### PR TITLE
[itv] Prefer last matching featureset

### PR DESCRIPTION
**Pull request by coletdjnz**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [X] At least skimmed through [adding new extractor tutorial](https://github.com/ytdl-org/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/ytdl-org/youtube-dl#youtube-dl-coding-conventions) sections
- [X] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [X] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [X] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [X] Bug fix
- [ ] Improvement
- [ ] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

Explanation of your *pull request* in arbitrary form goes here. Please make sure the description explains the purpose and effect of your *pull request* and is worded well enough to be understood. Provide as much context and examples as possible.

Bug fix for https://github.com/yt-dlp/yt-dlp/issues/986

@sleaux-meaux if you can confirm this fixes the issue that would be great thanks.

### CODE REVIEW

File: yt_dlp/extractor/itv.py [hunk 1]
```diff
@@ -113,11 +113,16 @@ def _call_api(self, video_id, playlist_url, headers, platform_tag, featureset, f
...
+        featureset_subs = None
+        # Prefer last matching featureset
+        # See: https://github.com/yt-dlp/yt-dlp/issues/986
+        for platform_tag, featuresets in variants.items():
+            for featureset in featuresets:
+                if 'outband-webvtt' in (featureset or []):
+                    platform_tag_subs = platform_tag
+                    featureset_subs = featureset
+                    break
         if platform_tag_subs or featureset_subs:
```
Comment (sleaux-meaux) [inline yt_dlp/extractor/itv.py:126]: Better to use `and` to make it clear both `platform_tag_subs` and `featureset_subs` are needed for API call

Comment (sleaux-meaux) [reply yt_dlp/extractor/itv.py:126]: Looks OK otherwise

Comment (coletdjnz) [reply yt_dlp/extractor/itv.py]: good catch, thanks

File: yt_dlp/extractor/itv.py [hunk 2]
```diff
@@ -113,12 +113,17 @@ def _call_api(self, video_id, playlist_url, headers, platform_tag, featureset, f
 
     def _get_subtitles(self, video_id, variants, ios_playlist_url, headers, *args, **kwargs):
         subtitles = {}
-        platform_tag_subs, featureset_subs = next(
-            ((platform_tag, featureset)
-             for platform_tag, featuresets in variants.items() for featureset in featuresets
-             if try_get(featureset, lambda x: x[2]) == 'outband-webvtt'),
-            (None, None))
-        if platform_tag_subs or featureset_subs:
```
Comment (pukkandan) [inline yt_dlp/extractor/itv.py:121]: why not just
```diff
         platform_tag_subs, featureset_subs = next(
             ((platform_tag, featureset)
-             for platform_tag, featuresets in variants.items() for featureset in featuresets
+             for platform_tag, featuresets in reversed(variants.items()) for featureset in featuresets
              if try_get(featureset, lambda x: x[2]) == 'outband-webvtt'),
             (None, None))
-        if platform_tag_subs or featureset_subs:
+        if platform_tag_subs and featureset_subs:
```
It is simpler and the intend to use the last item is clear from `reversed`

Comment (coletdjnz) [reply yt_dlp/extractor/itv.py:121]: yeah, sure

**Comment by sleaux-meaux** [comment]
Works

**Pull request merged**